src/main/java/uk/gov/pay/api/service/SearchRefundsService.java

```diff
@@ -41,4 +67,39 @@ public Response getAllRefunds(Account account, RefundsParams params) {
         queryParams.put(DISPLAY_SIZE, Optional.ofNullable(params.getDisplaySize()).orElse(DEFAULT_DISPLAY_SIZE));
         return queryParams;
     }
+
+    private Response getSearchResponse(Account account, Map<String, String> queryParams) {
+        String url = connectorUriGenerator.refundsURIWithParams(account, queryParams);
+        Response connectorResponse = client
+                .target(url)
+                .request()
+                .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
+                .get();
+        LOGGER.info("response from connector for refunds search: {}", connectorResponse);
+        if (connectorResponse.getStatus() == SC_OK) {
+            return processResponse(connectorResponse);
+        }
+        throw new SearchRefundsException(connectorResponse);
+    }
+
+    private Response processResponse(Response connectorResponse) {
+        try {
+            SearchRefundsResponse searchResponse = connectorResponse.readEntity(SearchRefundsResponse.class);
+            List<RefundForSearchRefundsResult> results = searchResponse.getRefunds()
+                    .stream()
+                    .map(refund -> RefundForSearchRefundsResult.valueOf(refund,
+                            publicApiUriGenerator.getPaymentURI(refund.getChargeId()),
+                            publicApiUriGenerator.getRefundsURI(refund.getChargeId(), refund.getRefundId())))
+                    .collect(Collectors.toList());
+
+            HalRepresentation.HalRepresentationBuilder halRepresentation = HalRepresentation
+                    .builder()
+                    .addProperty("results", results);
+            return Response.ok().entity(
+                    paginationDecorator.decoratePagination(halRepresentation, searchResponse, REFUNDS_PATH).build().toString())
+                    .build();
+        } catch (ProcessingException ex) {
+            throw new SearchRefundsException(ex);
```

Comment:
Same comment as the previous about catching the ProcessingException. If you really need to catch it, it's best to catch it around the `connectorResponse.readEntity(SearchRefundsResponse.class)` line rather around the entire code block as it makes it clearer which line can potentially throw the exception.

Reply:
This still needs to be addressed

Reply:
Ah sorry, missed that